clic/include/statistics.hpp

```diff
@@ -89,30 +89,144 @@ compute_statistics(const Device::Pointer & device, Array::Pointer & intensity, A
   std::iota(labels_list.begin(), labels_list.end(), offset);
   region_props["label"] = labels_list;
 
-  // // 1-6 bbox x, y, z, width, height, depth
-  // tier1::crop(device, min_per_label, result_vector, offset, 10, 0);
-  // bbox_min_x = pull(result_vector)[0];
-  // region_props['bbox_min_x'] = bbox_min_x;
-
-  // tier1::crop(device, min_per_label, result_vector, offset, 12, 0);
-  // bbox_min_y = pull(result_vector)[0];
-  // region_props['bbox_min_y'] = bbox_min_y;
-
-  // tier1::crop(device, min_per_label, result_vector, offset, 14, 0);
-  // bbox_min_z = pull(result_vector)[0];
-  // region_props['bbox_min_z'] = bbox_min_z;
-
-  // tier1::crop(device, max_per_label, result_vector, offset, 11, 0);
-  // bbox_max_x = pull(result_vector)[0];
-  // region_props['bbox_max_x'] = bbox_max_x;
-
-  // tier1::crop(device, max_per_label, result_vector, offset, 13, 0);
-  // bbox_max_y = pull(result_vector)[0];
-  // region_props['bbox_max_y'] = bbox_max_y;
-
-  // tier1::crop(device, max_per_label, result_vector, offset, 15, 0);
-  // bbox_max_z = pull(result_vector)[0];
-  // region_props['bbox_max_z'] = bbox_max_z;
+  // 1-6 bbox x, y, z, width, height, depth
+  tier1::crop(device, min_per_label, result_vector, offset, 10, 0);
+  std::vector<float> bbox_min_x = result_vector->as_vector<float>();
+  region_props["bbox_min_x"] = bbox_min_x;
+
+  tier1::crop(device, min_per_label, result_vector, offset, 12, 0);
+  std::vector<float> bbox_min_y = result_vector->as_vector<float>();
+  region_props["bbox_min_y"] = bbox_min_y;
+
+  tier1::crop(device, min_per_label, result_vector, offset, 14, 0);
+  std::vector<float> bbox_min_z = result_vector->as_vector<float>();
+  region_props["bbox_min_z"] = bbox_min_z;
+
+  tier1::crop(device, max_per_label, result_vector, offset, 11, 0);
+  std::vector<float> bbox_max_x = result_vector->as_vector<float>();
+  region_props["bbox_max_x"] = bbox_max_x;
+
+  tier1::crop(device, max_per_label, result_vector, offset, 13, 0);
+  std::vector<float> bbox_max_y = result_vector->as_vector<float>();
+  region_props["bbox_max_y"] = bbox_max_y;
+
+  tier1::crop(device, max_per_label, result_vector, offset, 15, 0);
+  std::vector<float> bbox_max_z = result_vector->as_vector<float>();
+  region_props["bbox_max_z"] = bbox_max_z;
+
+  // Calculate bbox width, height, depth
+  std::vector<float> bbox_width(nb_labels - offset);
+  std::vector<float> bbox_height(nb_labels - offset);
+  std::vector<float> bbox_depth(nb_labels - offset);
+
+  for (size_t i = 0; i < bbox_width.size(); ++i) {
+    bbox_width[i] = bbox_max_x[i] - bbox_min_x[i] + 1;
+    bbox_height[i] = bbox_max_y[i] - bbox_min_y[i] + 1;
+    bbox_depth[i] = bbox_max_z[i] - bbox_min_z[i] + 1;
+  }
+
+  region_props["bbox_width"] = bbox_width;
+  region_props["bbox_height"] = bbox_height;
+  region_props["bbox_depth"] = bbox_depth;
+
+  // Minimum and maximum intensity
+  tier1::crop(device, min_per_label, result_vector, offset, 8, 0);
```

Comment:
Would this work?

Reply:
nope, 

- `crop` need 6 parameters `start_x, start_y, start_z, width, height, depth`, CLIc do not have default values
- GPU Arrays do not have `->as_vector<float>();` methods

Reply:
> GPU Arrays do not have ->as_vector<float>(); methods

How does ´pull´ work in CLIC?  (sorry for my ignorance)
I found this:
```
gpu_output->read(output.data());
````

But, output is a std::array.

Reply:
This is correct.

Here `output` can be a `std::array` if you know already the number of element the kernel return, or a `std::vector` (more adapted), you simply have to create it before the pull:

```
std::vector<float> output(gpu_output->size());
```

Warning, the vector type and the gpu buffer type must be the same (float, for dType::FLOAT) etc.